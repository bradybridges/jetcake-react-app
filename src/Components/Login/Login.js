import React, { Component } from 'react';
import './Login.scss';
import * as firebase from 'firebase';
import 'firebase/auth';

export default class Login extends Component {
  state = {
    email: "",
    password: "",
    error: null,
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleLogin = async () => {
    const { email, password } = this.state;
    const { toggleShowLogin } = this.props;
    try {
      await firebase.auth().signInWithEmailAndPassword(email, password);
      toggleShowLogin();
    } catch(error) {
      this.setState({ error, email: "", password: "" });
      console.error(error);
    }
  }

  render() {
    const { toggleShowLogin } = this.props;
    if(this.state.error) {
      setTimeout(() => {
        this.setState({ error: null });
      }, 2500);
    }
    return (
      <section className="login-form">
        {this.state.error && <p id="error-msg">{this.state.error.message}</p>}
        <label>Email</label>
        <input 
          name="email" 
          type="text" 
          value={this.state.email} 
          onChange={this.handleChange}
        />
        <label>Password</label>
        <input 
          name="password" 
          type="password" 
          value={this.state.password} 
          onChange={this.handleChange}
        />
        <button onClick={this.handleLogin}>Login</button>
        <button onClick={toggleShowLogin}>Cancel</button>
      </section>
    )
  }
}

