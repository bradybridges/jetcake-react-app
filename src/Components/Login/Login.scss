.login-form {
  position: absolute;
  z-index: 5;
  top: 25%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  left: 50%;
  margin-left: -200px;
  width: 400px;
  background-color: blueviolet;
  border-radius: 4px;
  min-height: 200px;
  padding: 10px;
}

.input-field {
  width: 200px;
  height: 35px;
  border-radius: 4px;
  margin: 10px 0;
}

.input-label {
  color: white;
  font-size: 1.25em;
}

.form-btn {
  width: 100px;
  height: 35px;
  border-radius: 4px;
  transition: transform .2s;
  &:hover {
    transform: scale(1.05);
    cursor: pointer;
  }
}

#error-msg {
  color: white;
  font-weight: bold;
}

.form-btn-container {
  margin-top: 10px;
}