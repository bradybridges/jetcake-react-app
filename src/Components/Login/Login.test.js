import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login';

describe('Login', () => {
  let wrapper;
  const mockToggleShowLogin = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Login toggleShowLogin={mockToggleShowLogin} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange should update state of email or password', () => {
    const mockEmailEvent = {
      target: { name: 'email', value: 'test@test.com' },
    };
    const mockPasswordEvent = {
      target: { name: 'password', value: 'secretpassword' },
    };
    expect(wrapper.state()).toEqual({ email: '', password: '', error: null });
    wrapper.instance().handleChange(mockEmailEvent);
    expect(wrapper.state('email')).toEqual(('test@test.com'));
    wrapper.instance().handleChange(mockPasswordEvent);
    expect(wrapper.state('password')).toEqual('secretpassword');
  });
});
