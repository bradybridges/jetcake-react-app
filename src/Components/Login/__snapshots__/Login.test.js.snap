// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <label
    className="input-label"
  >
    Email
  </label>
  <input
    className="input-field"
    name="email"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    className="input-label"
  >
    Password
  </label>
  <input
    className="input-field"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <section
    className="form-btn-container"
  >
    <button
      className="form-btn"
      type="submit"
    >
      Login
    </button>
    <button
      className="form-btn"
      onClick={[Function]}
    >
      Cancel
    </button>
  </section>
</form>
`;
